modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributor.java

```diff
@@ -84,31 +83,34 @@ public Map<String, Object> getParameters(
 					RequestBackedPortletURLFactoryUtil.create(
 						ddmFormFieldRenderingContext.getHttpServletRequest());
 
+				Map<String, String[]> parameterMap = new HashMap<>();
+
+				for (Object object :
+						(Object[])ddmFormField.getProperty("parameters")) {
+
+					String parameter = (String)object;
+
+					String[] parameters = parameter.split(StringPool.EQUAL);
+
+					parameterMap.put(
+						parameters[0], new String[] {parameters[1]});
+				}
+
 				return PortletURLBuilder.create(
 					requestBackedPortletURLFactory.createActionURL(
 						GetterUtil.getString(
 							((Object[])ddmFormField.getProperty("portletId"))
 								[0]))
 				).setParameters(
-					HashMapBuilder.put(
+					HashMapBuilder.putAll(
+						parameterMap
```

Comment:
@JanisZhang Since you have to create a HashMap to populate most of the parameters, please just put the `mvcRenderCommandName` directly into that map, and pass the map directly into `setParameters`. This way we create one less map.